test/e2e/dynamic_resource_e2e_test.go

```diff
@@ -3,103 +3,164 @@ package e2e
 import (
 	"strings"
 
-	"github.com/stretchr/testify/require"
 	corev1 "k8s.io/api/core/v1"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/client-go/dynamic"
 
 	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
 	"github.com/operator-framework/operator-lifecycle-manager/pkg/api/apis/operators/v1alpha1"
+	"github.com/operator-framework/operator-lifecycle-manager/pkg/api/client/clientset/versioned"
 	"github.com/operator-framework/operator-lifecycle-manager/pkg/controller/registry"
+	"github.com/operator-framework/operator-lifecycle-manager/pkg/lib/operatorclient"
 	"github.com/operator-framework/operator-lifecycle-manager/test/e2e/ctx"
 )
 
-var _ = Describe("Dynamic Resource", func() {
-	It("resolve prometheus API", func() {
+var _ = Describe("Subscriptions create required objects from Catalogs", func() {
 
-		defer cleaner.NotifyTestComplete(GinkgoT(), true)
+	var (
+		c             operatorclient.ClientInterface
+		crc           versioned.Interface
+		dynamicClient dynamic.Interface
+		deleteOpts    *metav1.DeleteOptions
+	)
 
-		c := newKubeClient(GinkgoT())
-		crc := newCRClient(GinkgoT())
-		dynamicClient := ctx.Ctx().DynamicClient()
+	BeforeEach(func() {
+		c = newKubeClient()
+		crc = newCRClient()
+		dynamicClient = ctx.Ctx().DynamicClient()
+	})
+
+	AfterEach(func() {
+		deleteOpts = &metav1.DeleteOptions{}
```

Comment:
Why is it important to assign `deleteOpts` here? Should this be happening in `BeforeEach`?

Reply:
It was because `deleteOpts` are only used for deleting resources (ns, catalog source) and which were only used in `AfterEach`

Reply:
Keep in mind that `AfterEach` closures run from innermost to outermost in situations where you have nested test scopes, like this one. So this assignment is happening AFTER it is being used elsewhere. I suspect that the client's Delete method allows nil for its DeleteOptions argument and it coincidentally working, but I don't think this is doing what you expect. Since it never changes, why not just assign it a value at the time it is declared?

Reply:
yep, you are right. Moved the initialization to `BeforeEach` block